2-mandatory/3-groceries.js

```diff
@@ -28,13 +28,37 @@ Exercise 1:
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
 
+for (let day in weeklyMealPlan) {
+  for (let ingredient of weeklyMealPlan[day]) {
+    if (weeklyGroceriesToBuy.indexOf(ingredient) === -1) {
+      weeklyGroceriesToBuy.push(ingredient);
+    }
+  }
+}
+
+//console.log(weeklyGroceriesToBuy);
+
+weeklyGroceriesToBuy = [...new Set(Object.values(weeklyMealPlan).flat())];
+
+//console.log(weeklyGroceriesToBuy);
 /*
 Exercise 2:
   Loop through your list again, but now only collect the weekend items into the weekendGroceriesToBuy array.
 */
 // Gather weekend item names into this array
 let weekendGroceriesToBuy = [];
 
+for(let day in weeklyMealPlan){
+  if(day === ('saturday' || 'sunday')){
```

Comment:
This is very optimistic, but it won't work. Can you look into why and explain why the following are true/false?

1. `'saturday' === ('saturday' || 'sunday')` results in true
2. `'sunday' === ('saturday' || 'sunday')` results in false
3. `'anything other than saturday' === ('saturday' || 'sunday')` results in false

let me know if you have questions or need explanation

Reply:
and why did tests _not fail_?